src/data_handler/data_handler.cpp

```diff
@@ -495,6 +492,62 @@ int perform_wavelet_denoising (double *data, int data_len, char *wavelet, int de
     }
     return (int)BrainFlowExitCodes::STATUS_OK;
 }
+  
+int perform_windowing (
+    double *data, int data_len, int window_function, double *output_data)
+{
+    if ((data == NULL) || (data_len <= 0) || (window_function < 0) || (output_data == NULL))
+    {
+        data_logger->error ("Pleck check the arguments: data must be >= 0, data_len > 0, window_function >= 0 and output_data cannot be empty. data:{} , "
+                            "window_function:{}, data_len:{}, output_data:{}",
+                            *data, window_function, data_len, *output_data);
+        return (int)BrainFlowExitCodes::INVALID_ARGUMENTS_ERROR;
+    }
+
+    double *windowed_data = new double[data_len];
+    try {
+        if (data == 0)
+        {
+            throw new std::runtime_error ("invalid x and f arguments");
+        }
+
+        // from https://www.edn.com/windowing-functions-improve-fft-results-part-i/
+        switch (static_cast<WindowFunctions> (window_function))
+        {
+            case WindowFunctions::NO_WINDOW:
+                windowed_data = no_window_function(data, data_len);
```

Comment:
setup clang-format plugin to enforce the same c++ code style across all files and developers https://brainflow.readthedocs.io/en/stable/BrainFlowDev.html#code-style